connect-inject/endpoints_controller.go

```diff
@@ -208,13 +226,26 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
 		tags = append(tags, strings.Split(raw, ",")...)
 	}
 
+	status, _, err := getReadyStatusAndReason(&pod)
```

Comment:
we dont use the "reason" field here because setting the Message field in the check on registration causes the Notes/Output field to get jumbled the next time we issue an update which occurs a couple ms later. It's fine and only has the effect of there being no Notes field until the UpdateTTL() is issues a couple ms later.

Reply:
The Notes field gets set and you cannot override it with later update.
So you end up with:
```
 Notes:  "k8s probe failed"
 Output: "k8s probe healthy"
```

Reply:
Agreed, I'll add a comment about it!